apps/website-25/src/components/courses/CertificateLinkCard.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  addQueryParam,
+  Auth,
+  Card, CTALinkOrButton, ProgressDots, useAuthStore,
+} from '@bluedot/ui';
+import useAxios from 'axios-hooks';
+import React from 'react';
+import { FaAward } from 'react-icons/fa6';
+import { ErrorView } from '@bluedot/ui/src/ErrorView';
+import { GetCourseRegistrationResponse } from '../../pages/api/course-registrations/[courseId]';
+import { ROUTES } from '../../lib/routes';
+import { RequestCertificateRequest, RequestCertificateResponse } from '../../pages/api/certificates/request';
+
+type CertificateLinkCardProps = {
+  courseId: string;
+};
+
+const CertificateLinkCard: React.FC<CertificateLinkCardProps> = ({
+  courseId,
+}) => {
+  const auth = useAuthStore((s) => s.auth);
+
+  if (!auth) {
+    return (
+      <Card
+        title="Your Certificate"
+        subtitle="Create a free account to collect course certificates."
+        className="container-lined p-8 bg-white"
+      >
+        <CTALinkOrButton
+          url={typeof window === 'undefined' ? ROUTES.login.url : addQueryParam(ROUTES.login.url, 'redirect_to', window.location.pathname)}
+          variant="primary"
+        >
+          Log in
+        </CTALinkOrButton>
+      </Card>
+    );
+  }
+
+  return <CertificateLinkCardAuthed courseId={courseId} auth={auth} />;
+};
+
+const CertificateLinkCardAuthed: React.FC<CertificateLinkCardProps & { auth: Auth }> = ({
+  courseId,
+  auth,
+}) => {
+  const [{ data, loading, error }, refetch] = useAxios<GetCourseRegistrationResponse>({
+    method: 'get',
+    url: `/api/course-registrations/${courseId}`,
+    headers: {
+      Authorization: `Bearer ${auth?.token}`,
+    },
+  });
+
+  const [{ loading: certificateRequestLoading, error: certificateRequestError }, refetchCertificateRequest] = useAxios<RequestCertificateResponse, RequestCertificateRequest>({
+    method: 'post',
+    url: '/api/certificates/request',
+    data: {
+      courseId,
+    },
+    headers: {
+      Authorization: `Bearer ${auth?.token}`,
+    },
+  }, { manual: true });
+
+  const requestCertificate = async () => {
+    try {
+      await refetchCertificateRequest();
+      await refetch();
+    } catch {
+      // Ignore, handled already by useAxios
+    }
+  };
+
+  if (error || certificateRequestError) {
+    return (
+      <Card
+        title="Your Certificate"
+        className="container-lined p-8 bg-white"
+      >
+        <div className="flex flex-col gap-4">
+          <ErrorView error={error || certificateRequestError} />
```

Comment:
Yeah, this is a little heavyweight for right here. But I think this is acceptable for now.